webapp/src/ts/modules/tasks/tasks.component.ts

```diff
@@ -155,20 +153,21 @@ export class TasksComponent implements OnInit, OnDestroy {
 
       this.tasksActions.setTasksList(hydratedTasks);
 
-      if (!this.tasksLoaded) {
-        this.tasksActions.setTasksLoaded(true);
-      }
     } catch (exception) {
       console.error('Error getting tasks for all contacts', exception);
       this.errorStack = exception.stack;
       this.loading = false;
       this.hasTasks = false;
       this.tasksActions.setTasksList([]);
     } finally {
+      const performanceName = this.tasksLoaded ? 'tasks:refresh' : 'tasks:load';
       this.trackPerformance?.stop({
-        name: this.tasksLoaded ? 'tasks:refresh' : 'tasks:load',
-        recordApdex: !this.tasksLoaded,
+        name: performanceName,
+        recordApdex: true,
```

Comment:
Here, we only want to record the apdex for `tasks:load`

Reply:
Why would we want to record only `tasks:load` from there?  `recordApdex: true` allows us to record both apdex from one place.

Reply:
is the `task:refresh` only called when task tab loads? or triggered by something else?

Reply:
First time we record `tasks:load`, subsequent visits we record `tasks:refresh`

https://github.com/medic/cht-core/assets/2597305/9215c155-f38f-4258-b277-3633c1a38522

Reply:
Got it! Thanks for the video, then this makes sense